pkg/controller/core/workload_controller.go

```diff
@@ -305,14 +282,7 @@ func (r *WorkloadReconciler) Update(e event.UpdateEvent) bool {
 		if !r.cache.AddOrUpdateWorkload(wlCopy) {
 			log.V(2).Info("ClusterQueue for workload didn't exist; ignored for now")
 		}
-	case prevStatus == admitted && status == cancellingAdmission:
-		// The workload will be requeued when handling transitioning
-		// from cancellingAdmission to pending.
-		// When the workload is in the cancellingAdmission status, the only
-		// transition possible, triggered by the workload controller, is to the
-		// pending status. Scheduler is only able to re-admit the workload once
-		// requeued after reaching the pending status.
-	case prevStatus == cancellingAdmission && status == pending:
+	case status == pending:
```

Comment:
```suggestion
	case prevStatus == admitted && status == pending:
```

it might be functionally the same, but it improves the readability, by clearly stating the transition that we care about.